### PR TITLE
fix(ilp): fix no periodic commits for WAL tables in ILP when table constantly written

### PR DESCRIPTION
**Pull request by ideoma**
When ILP writes data to WAL table without pauses then the commit only happens when `max uncommitted rows` is exceeded. Instead, it should commit with the default setting at least 2 times per second.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpCommitFuzzTest.java [hunk 1]
```diff
@@ -96,6 +100,34 @@ public void testCommitIntervalBasedFraction() throws Exception {
...
+    public void testCommitIntervalBasedFractionNoPauses() throws Exception {
+
+        configOverrideMaxUncommittedRows(500_000);
+
+        // idle table commit after 5 mins inactivity -> test would timeout
+        maintenanceInterval = 300_000_000;
+        minIdleMsBeforeWriterRelease = 300_000_000;
+
+        // time based commit every 0.5 seconds (50% of 1 sec commit lag) -> should commit rows -> make test pass
+        configOverrideO3MaxLag(500_000);
```
Comment (eugenels) [inline core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpCommitFuzzTest.java:113]: to tune the commit and heartbeat interval, use: O3MinLag

**Comment by eugenels** [comment]
`AbstractLineTcpReceiverTest` has a hard-coded heartbeat interval.
to be able to configure it must be:
```
        @Override
        public long getHeartbeatInterval() {
            return lineConfiguration.getCommitInterval();
        }
```

**Comment by eugenels** [comment]
`LineTcpWriterJob` does not commit under load as well:
```
    @Override
    public boolean run(int workerId, @NotNull RunStatus runStatus) {
        assert this.workerId == workerId;
        boolean busy = drainQueue();
        // while ILP is hammering the database via multiple connections the writer
        // is likely to be very busy so commitTables() will run infrequently
        // commit should run regardless the busy flag but has to finish quickly
        // idea is to store the tables in a heap data structure being the tables most
        // desperately need a commit on the top
        if (!busy) {
            commitTables();
            tickWriters();
        }
        return busy;
    }
```

**Comment by ideoma** [comment]
> `LineTcpWriterJob` does not commit under load as well:
> 
> ```
>     @Override
>     public boolean run(int workerId, @NotNull RunStatus runStatus) {
> ```

In the test scenario, there is no **load** really. It's just lines spaced every 10ms, 100 lines per second. And in the case will not happen until max uncommitted rows reached for the table, e.g. 5000s

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|

**Review by glasstiger — APPROVED** [review]

**Pull request merged**